handlers/secrets.go

```diff
@@ -75,3 +81,86 @@ func makeSecretsArray(c *client.Client, secretNames []string) ([]*swarm.SecretRe
 
 	return values, nil
 }
+
+func secretGetHandler(c *client.Client, w http.ResponseWriter, r *http.Request) {
+	secrets, err := c.SecretList(context.Background(), types.SecretListOptions{})
+
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+
+		return
+	}
+
+	var results []requests.Secret
+
+	for _, s := range secrets {
+		results = append(results, requests.Secret{Name: s.Spec.Name})
+	}
+
+	bs, marshalErr := json.Marshal(results)
+
+	if marshalErr != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+
+		return
+	}
+
+	_, writeErr := w.Write(bs)
+
+	if writeErr != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+
+		return
+	}
+
+	w.WriteHeader(http.StatusOK)
+}
+
+func secretPostHandler(c *client.Client, w http.ResponseWriter, r *http.Request) {
```

Comment:
@LucasRoesler what about reading the request in a generic way and then creating methods like createSecret / deleteSecret etc which do not directly interact with the http.* namespace?

Reply:
Good idea

Reply:
I'm happy for that to be incremental, so doesn't have to be in this initial PR which is about getting something working that can be maintained easily.

Reply:
Ok, so for now I'm leaving it, and in next iteration I'll refactor it

Reply:
I already refactored it